backend/remote/backend.go

```diff
@@ -255,8 +265,29 @@ func (b *Remote) Configure(obj cty.Value) tfdiags.Diagnostics {
 					`Terraform Enterprise client: %s.`, err,
 			),
 		))
+		return diags
 	}
 
+	// Check if the organization exists.
+	_, err = b.client.Organizations.Read(context.Background(), b.organization)
+	if err != nil {
+		if err == tfe.ErrResourceNotFound {
+			err = fmt.Errorf("organization %s does not exist", b.organization)
+		}
+		diags = diags.Append(tfdiags.Sourceless(
```

Comment:
It seems like errors here are essentially validation for the `organization` argument, so perhaps better to return a `tfdiags.AttributeValue` result here instead so that the error can point directly to that argument, rather than to the block as a whole?

I can imagine this returning errors that are not strictly user config errors, like the TFE API being totally unreachable, but I think the detail message below makes it clear that we were making a call out here so the resulting full rendered error message should still be clear.

Reply:
Updated